test/util.py

```diff
@@ -34,3 +34,17 @@ def decorated(*args, **kwds):
             return f(*args, **kwds)
     decorated.__name__ = f.__name__
     return decorated
+
+class temporary_file():
+
+    def __init__(self):
+        pass
+
+    def __enter__(self):
+        self.temp_file = tempfile.NamedTemporaryFile(delete=False)
+        return self.temp_file
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        if self.temp_file and os.path.exists(self.temp_file.name):
+            self.temp_file.close()
+            os.remove(self.temp_file.name)
```

Comment:
I don't see how this class is significantly different to `tempfile.NamedTemporaryFile`?

Reply:
Basically it drops the _O_TEMPORARY usage internally specified inside the NamedTemporaryFile when the delete=True. Under the hood, _open with _O_TEMPORARY is passing FILE_FLAG_DELETE_ON_CLOSE into Win32 API CreateFile and others cannot open the same file unless they are explicitly using FILE_SHARE_DELETE.
https://docs.microsoft.com/en-us/windows/desktop/api/fileapi/nf-fileapi-createfilea

So my proposed workaround is to remove this unwanted effect since the code under test needs to touch the file but without knowing this _O_TEMPORARY (or FILE_SHARE_DELETE) details. Any thoughts on this?

Reply:
These internals are not obvious from this patch and not even mentioned in the Python docs. They heavily rely on the implementation details (which could change anytime).

Hence I think my earlier proposal to create a `TemporaryDirectory` is the preferred approach.

Reply:
That makes sense. I go use in_temporary_directory and mkstemp to replace the current implementation.